### PR TITLE
Jurisdiction auto selection on assignment page

### LINKED ISSUE #990: Jurisdiction Auto-Selection on Jurisdiction Assignment Page
It should be possible to:

- [x] 1. Have a config (env var) that identifies a jurisdiction metadata identifier (e.g. risk_2019) to use to jurisdiction auto selection
- [x] 2. Have a jurisdiction metadata reducer, perhaps built on [reducer factory](https://github.com/OpenSRP/opensrp-web/tree/master/packages/opensrp-reducer-factory) if possible.  This reducer will be responsible for storing jurisdiction metadata and making it possible to select by: 
    - identifier
    - jurisdiction id
- [x] 3. Extend [Jurisdiction Assignment Table](https://github.com/onaio/reveal-frontend/blob/master/src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx) so that it loads with pre-selected jurisdictions based on the jurisdiction metadata in (1) above and a threshold e.g. auto select all jurisdictions whose risk_2019 is more than 85%

### PR DESCRIPTION
**Pull request by KipSigei**
- Auto select jurisdictions based on metadata risk and risk threshold
- Add an `Assignment Type` column to show whether a jurisdiction was auto selected or not

closes #990

### CODE REVIEW

File: src/configs/lang.ts [hunk 1]
```diff
@@ -720,6 +720,11 @@ export const COULD_NOT_LOAD_JURISDICTION_HIERARCHY = translate(
 
 export const STRUCTURES_COUNT = translate('STRUCTURES_COUNT', 'Structures Count');
 
+export const ASSIGNMENT_TYPE = translate('ASSIGNMENT_TYPE', 'Assignment Type');
+
+export const AUTO = translate('AUTO', 'Auto');
```
Comment (moshthepitt) [inline src/configs/lang.ts:725]: Lets change the test to "Auto-selection"

File: src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx [hunk 2]
```diff
@@ -67,12 +83,34 @@ export type JurisdictionAssignmentViewFullProps = JurisdictionAssignmentViewProp
...
+    loadJurisdictionsMetadata(
+      JURISDICTION_METADATA_RISK,
+      OpenSRPService,
+      fetchJurisdictionsMetadataCreator
+    )
+      .then(() => {
+        setLoading(false);
+      })
+      .catch(error => {
+        setLoading(false);
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx:109]: Lets `setLoading(false)` in the `finally` callback

File: src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx [hunk 3]
```diff
@@ -67,12 +83,34 @@ export type JurisdictionAssignmentViewFullProps = JurisdictionAssignmentViewProp
...
+        return error;
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionAssignmentView/index.tsx:110]: Should we use `displayError` here?

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 4]
```diff
@@ -188,8 +217,8 @@ const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
     currentPage,
     pages,
   };
-
   const data = currentChildren.map(node => {
+    const isLeafNode: boolean = leafNodes.map(leaf => leaf.model.id).includes(node.model.id);
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:221]: Is it possible to replace this with `node.hasChildren()`?

File: src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx [hunk 5]
```diff
@@ -202,10 +231,20 @@ const JurisdictionTable = (props: JurisdictionSelectorTableProps) => {
...
       <NodeCell key={`${node.model.id}-jurisdiction`} node={node} baseUrl={baseUrl} />,
+      node.model.meta.selected && isLeafNode
+        ? jurisdictionMetaIds.length &&
+          jurisdictionMetaIds.includes(node.model.id) &&
+          !leafNodes.filter(value => existingAssignments.includes(value.model.id)).length
+          ? AUTO
+          : leafNodes.filter(value => existingAssignments.includes(value.model.id)).length &&
+            existingAssignments.includes(node.model.id)
+          ? EXISTING
+          : 'Manual'
```
Comment (moshthepitt) [inline src/containers/pages/JurisdictionAssignment/JurisdictionTable/index.tsx:242]: Lets get this from the lang strings module

File: src/services/opensrp/index.ts [hunk 6]
```diff
@@ -72,7 +72,7 @@ export class OpenSRPService extends OpenSRPServiceWeb {
   }
 
   public async readFile(
-    id: string | number,
+    id?: string | number,
```
Comment (moshthepitt) [inline src/services/opensrp/index.ts:75]: Why is this one necessary?

File: src/store/ducks/opensrp/hierarchies/index.ts [hunk 7]
```diff
@@ -396,3 +396,19 @@ export const getAllSelectedNodes = () =>
     });
     return nodesList;
   });
+
+export const getLeafNodes = () =>
```
Comment (moshthepitt) [inline src/store/ducks/opensrp/hierarchies/index.ts:400]: Could we add a docstring here?

Comment (moshthepitt) [reply src/store/ducks/opensrp/hierarchies/index.ts:400]: Lets also add a test for `getLeafNodes`

File: src/store/ducks/opensrp/jurisdictionsMetadata/index.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
+import { AnyAction, Store } from 'redux';
```
Comment (moshthepitt) [inline src/store/ducks/opensrp/jurisdictionsMetadata/index.ts:1]: For such a simple reducer we should use [Reducer factory](https://github.com/OpenSRP/opensrp-web/tree/master/packages/opensrp-reducer-factory).  However, at this point I am fine of you create an issue for this to be done later.

File: src/services/opensrp/index.ts (hunk 6 above)
Comment (KipSigei) [reply src/services/opensrp/index.ts:75]: The jurisdictions settings endpoint doesn't need an id URL param so i had to make the id param optional

File: src/configs/lang.ts (hunk 1 above)
Comment (KipSigei) [reply src/configs/lang.ts]: addressed via https://github.com/onaio/reveal-frontend/pull/996/commits/ef166ce296e7f8067317a975812197cbdc973e54

**Comment by KipSigei** [comment]
> #### Let's look into:
> * Lets show a column of the risk for each jurisdiction
> * In my limited testing it feels like if you make selections it doesn't save to the server consistently

These have been addressed you can check once more

File: src/store/ducks/opensrp/jurisdictionsMetadata/index.ts (hunk 8 above)
Comment (moshthepitt) [reply src/store/ducks/opensrp/jurisdictionsMetadata/index.ts:1]: @KipSigei did you create the issue?

File: src/services/opensrp/index.ts (hunk 6 above)
Comment (moshthepitt) [reply src/services/opensrp/index.ts:75]: cool

File: src/store/ducks/opensrp/jurisdictionsMetadata/index.ts (hunk 8 above)
Comment (KipSigei) [reply src/store/ducks/opensrp/jurisdictionsMetadata/index.ts:1]: @moshthepitt https://github.com/onaio/reveal-frontend/issues/1012

**Comment by Naima-Bashir** [comment]
Need help testing this. It does not have the acceptance criterai. cc @p-netm @moshthepitt

**Pull request merged**